### PR TITLE
do not check test dirs in non pack integrations/scripts

### PR DESCRIPTION
**Pull request by orhovy**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/22809

## Description
When running demisto-sdk validate, going through scripts and integrations , inner folders of test_data wont be checked

### CODE REVIEW

File: demisto_sdk/commands/validate/file_validator.py [hunk 1]
```diff
@@ -589,6 +588,8 @@ def validate_all_files(self):
                     project_dir = os.path.join(root, inner_dir)
                     _, file_path = get_yml_paths_in_dir(project_dir, Errors.no_yml_file(project_dir))
                     if file_path:
+                        if get_matching_regex(file_path, TEST_REG):
```
Comment (bakatzir) [inline demisto_sdk/commands/validate/file_validator.py]: please add a comment to explain

File: demisto_sdk/commands/common/constants.py [hunk 2]
```diff
@@ -438,6 +438,10 @@ def invalid_v2_script_name(file_path):
 INTEGRATION_README_REGEX = r'{}{}/([^\\/]+)/README.md$'.format(CAN_START_WITH_DOT_SLASH, INTEGRATIONS_DIR)
 INTEGRATION_CHANGELOG_REGEX = r'{}{}/([^\\/]+)/CHANGELOG.md$'.format(CAN_START_WITH_DOT_SLASH, INTEGRATIONS_DIR)
 
+TEST_DATA_INTEGRATION_YML_REGEX = r'{}{}/([^\\/]+)/([^\\/]+)/([^\\/]+).yml$'.format(CAN_START_WITH_DOT_SLASH, INTEGRATIONS_DIR)
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py]: anything under `test_data` should be ignored. same for scripts
```suggestion
TEST_DATA_INTEGRATION_YML_REGEX = r'{}{}/([^\\/]+)/test_data/.*$'.format(CAN_START_WITH_DOT_SLASH, INTEGRATIONS_DIR)
```

File: demisto_sdk/commands/common/constants.py [hunk 3]
```diff
@@ -823,6 +828,11 @@ def invalid_v2_script_name(file_path):
     PACKS_SCRIPT_JS_REGEX
 ]
 
+TEST_REG = [
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py]: what about packs?

File: demisto_sdk/commands/validate/file_validator.py (hunk 1 above)
Comment (yaakovi) [inline demisto_sdk/commands/validate/file_validator.py]: you can avoid all regex use by checking if `test_data` is part of the file path

File: demisto_sdk/commands/common/constants.py (hunk 2 above)
Comment (orhovy) [reply demisto_sdk/commands/common/constants.py]: the name of the folder is not always test_data

File: demisto_sdk/commands/common/constants.py (hunk 3 above)
Comment (orhovy) [reply demisto_sdk/commands/common/constants.py]: added packs now too

File: demisto_sdk/commands/validate/file_validator.py (hunk 1 above)
Comment (orhovy) [reply demisto_sdk/commands/validate/file_validator.py]: added

**Comment by bakatzir** [comment]
@orhovy @yaakovi 
on second though I also thing we should not user regexes, but a constants of test dir names, e.g: `[test_data,tests_data,<integration_name>_test]` catch without considering of capital letters, and filter those directories out of the checks.

I believe it will catch the vast majority.

**Comment by yaakovi** [comment]
@bakatzir @orhovy 

`find -f . -- -type d -iname "*test*" | grep -v -E "venv|TestPlaybooks|.cache|docs"`

list of all test folders:
```
./Packs/rasterize/Integrations/rasterize/test_data
./Packs/FeedCofense/Integrations/FeedCofense/TestData
./Packs/Zscaler/Integrations/Zscaler/test_data
./Packs/FeedOffice365/Integrations/FeedOffice365/test_data
./Packs/Pcysys/Integrations/Pcysys/TestData
./Packs/FeedProofpoint/Integrations/FeedProofpoint/TestData
./Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/test_data
./Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators_test
./Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators_test/TestHelperFunctions
./Packs/ApiModules/Scripts/HTTPFeedApiModule/test_data
./Packs/ApiModules/Scripts/CSVFeedApiModule/test_data
./Packs/ApiModules/Scripts/JSONFeedApiModule/test_data
./Integrations/RedCanary/TestData
./Integrations/FireEyeHelix/test_data
./Integrations/AttackIQFireDrill/test_data
./Integrations/MicrosoftGraphFiles/test_data
./Integrations/RiskSense/TestData
./Integrations/Akamai_SIEM/Akamai_SIEM_test
./Integrations/Akamai_SIEM/Akamai_SIEM_test/TestCommandsFunctions
./Integrations/MicrosoftGraphGroups/test_data
./Integrations/FeedTAXII/FeedTAXII_test
./Integrations/FeedTAXII/FeedTAXII_test/TestCommands
./Integrations/FeedTAXII/FeedTAXII_test/StixDecodeTest
./Integrations/ZeroFox/TestData
./Integrations/Exabeam/test_data
./Integrations/WindowsDefenderAdvancedThreatProtection/test_data
./Integrations/Flashpoint/TestData
./Integrations/Lastline_v2/data_test
./Integrations/SymantecEndpointProtection_V2/test-data
./Integrations/GoogleChronicleBackstory/TestData
./Integrations/ThreatQ_v2/test_data
./Integrations/MISP_V2/test_data
./Integrations/Securonix/test_data
./Integrations/MicrosoftGraphMail/test_data
./Integrations/Whois/test_data
./Integrations/Kafka_V2/test_data
./Integrations/Accessdata/test_data
./Integrations/EDL/EDL_test
./Integrations/EDL/EDL_test/TestHelperFunctions
./Integrations/KennaV2/Tests_Data
./Integrations/SMIME_Messaging/test_data
./Integrations/CounterCraft/test_data
./Integrations/CrowdStrikeFalcon/test_data
./Integrations/MicrosoftGraphListener/test_data
./Integrations/PhishLabsIOC_DRP/test_data
./Integrations/ImageOCR/test_data
./Scripts/ParseCSV/TestData
./Scripts/DBotTrainTextClassifierV2/TestData
./Scripts/ImportMLModel/TestData
./Scripts/ReadPDFFileV2/test_data
./Scripts/Ping/test_data
./Scripts/YaraScan/test_data
./Scripts/RegexExtractAll/TestData
./Scripts/GetFieldsByIncidentType/TestData
./Scripts/StixParser/TestData
./Scripts/FileToBase64List/test_data
./Scripts/ConvertFile/test_data
./Scripts/ExtractIndicatorsFromWordFile/test_data
./Scripts/IncidentFields/TestData
./Scripts/ProvidesCommand/TestData
./Scripts/WordTokenizeTest
./Scripts/DBotPreprocessTextData/TestData
./Scripts/JSONFileToCSV/test_data
./Scripts/GetDockerImageLatestTag
./Scripts/DocumentationAutomation/test_data
./Scripts/ExportMLModel/TestData
./Scripts/ParseEmailFiles/test_data
./Scripts/UnzipFile/data_test
./Scripts/HTMLDocsAutomation/test_data
```

catching `TestData` and `test_data` will catch most of them.
other:
```
./Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators_test
./Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators_test/TestHelperFunctions
./Integrations/Akamai_SIEM/Akamai_SIEM_test
./Integrations/Akamai_SIEM/Akamai_SIEM_test/TestCommandsFunctions
./Integrations/FeedTAXII/FeedTAXII_test
./Integrations/FeedTAXII/FeedTAXII_test/TestCommands
./Integrations/FeedTAXII/FeedTAXII_test/StixDecodeTest
./Integrations/Lastline_v2/data_test
./Integrations/SymantecEndpointProtection_V2/test-data
./Integrations/EDL/EDL_test
./Integrations/EDL/EDL_test/TestHelperFunctions
./Integrations/KennaV2/Tests_Data
./Scripts/WordTokenizeTest
./Scripts/GetDockerImageLatestTag
./Scripts/UnzipFile/data_test
```

File: demisto_sdk/commands/common/constants.py [hunk 4]
```diff
@@ -438,6 +438,7 @@ def invalid_v2_script_name(file_path):
 INTEGRATION_README_REGEX = r'{}{}/([^\\/]+)/README.md$'.format(CAN_START_WITH_DOT_SLASH, INTEGRATIONS_DIR)
 INTEGRATION_CHANGELOG_REGEX = r'{}{}/([^\\/]+)/CHANGELOG.md$'.format(CAN_START_WITH_DOT_SLASH, INTEGRATIONS_DIR)
 
+
```
Comment (bakatzir) [inline demisto_sdk/commands/common/constants.py]: rm

File: demisto_sdk/commands/common/constants.py [hunk 5]
```diff
@@ -451,6 +452,7 @@ def invalid_v2_script_name(file_path):
                                                                              INTEGRATIONS_DIR)
 PACKS_INTEGRATION_REGEX = r'{}{}/([^/]+)/{}/([^/]+)\.yml'.format(CAN_START_WITH_DOT_SLASH, PACKS_DIR, INTEGRATIONS_DIR)
 PACKS_SCRIPT_YML_REGEX = r'{}{}/([^/]+)/{}/([^/]+)/\2\.yml'.format(CAN_START_WITH_DOT_SLASH, PACKS_DIR, SCRIPTS_DIR)
+
```
Comment (bakatzir) [inline demisto_sdk/commands/common/constants.py]: rm

File: demisto_sdk/commands/common/constants.py [hunk 6]
```diff
@@ -824,6 +826,7 @@ def invalid_v2_script_name(file_path):
     PACKS_SCRIPT_JS_REGEX
 ]
 
+
```
Comment (bakatzir) [inline demisto_sdk/commands/common/constants.py]: rm

File: demisto_sdk/commands/validate/file_validator.py [hunk 7]
```diff
@@ -559,6 +559,10 @@ def validate_all_files(self):
                             _, file_path = get_yml_paths_in_dir(os.path.normpath(project_dir),
                                                                 Errors.no_yml_file(project_dir))
                             if file_path:
+                                # check if the file_path is part of test_data yml
+                                if any(test_file in file_path.lower() for test_file in
```
Comment (bakatzir) [inline demisto_sdk/commands/validate/file_validator.py]: let's use a CONST.
say TESTS_DIRECTORIES = ['testdata', 'test_data', 'data_test']

File: demisto_sdk/commands/validate/file_validator.py [hunk 8]
```diff
@@ -589,6 +592,9 @@ def validate_all_files(self):
                     project_dir = os.path.join(root, inner_dir)
                     _, file_path = get_yml_paths_in_dir(project_dir, Errors.no_yml_file(project_dir))
                     if file_path:
+                        # check if the file_path is part of test_data yml
+                        if any(test_file in file_path.lower() for test_file in ['testdata', 'test_data', 'data_test']):
```
Comment (bakatzir) [inline demisto_sdk/commands/validate/file_validator.py]: use same CONST

**Comment by orhovy** [comment]
@bakatzir ran validate on master and none of the test files were checked.

**Pull request merged**